orchestrator/lib/coolfhir/contained.go

```diff
@@ -0,0 +1,98 @@
+package coolfhir
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+)
+
+var (
+	ErrCareTeamNotFound = errors.New("CareTeam not found in FHIR Bundle")
+	ErrInvalidReference = errors.New("invalid reference, expecting local reference")
+)
+
+type ContainedResource struct {
+	Resource
+	Raw json.RawMessage `json:"-"`
+}
+
+func (r *ContainedResource) UnmarshalJSON(data []byte) error {
+	if err := json.Unmarshal(data, &r.Resource); err != nil {
+		return err
+	}
+
+	r.Raw = data
+
+	return nil
+}
+
+func (r *ContainedResource) MarshalJSON() ([]byte, error) {
+	return r.Raw, nil
+}
+
+func UpdateContainedResource(contained json.RawMessage, id *fhir.Reference, resourceType string, resource any) (json.RawMessage, error) {
```

Comment:
a `fhir.Reference` already contains a type (although it might be nil). So just use the reference and remove the `id` parameter, or change the type of `id` to `string`.